theano/sandbox/gpuarray/neighbours.py

```diff
@@ -41,17 +52,24 @@ def make_node(self, ten4, neib_shape, neib_step):
 
         return Apply(self, [ten4, neib_shape, neib_step],
                      [GpuArrayType(broadcastable=(False, False),
-                                   dtype=ten4.type.dtype)()])
+                                   dtype=ten4.type.dtype,
+                                   context=self.context)()])
+
+    def get_context(self, node):
+        return self.context
+
+    def perform(self, node, inp, out, ctx):
+        raise MethodNotDefined()
 
     def c_code_cache_version(self):
-        return (9,1)
+        return (9,2)
 
     def c_headers(self):
         return ['cuda.h', '<gpuarray/extension.h>', '<numpy_compat.h>',
                 '<gpuarray/ext_cuda.h>']
 
     def c_compiler(self):
-        return NVCC_compiler
+        return NVCC_compiler(self.context)
```

Comment:
I don't either, but the proper fix is to change the ops to not require the use of NVCC_compiler.  It has always been a temporary transitional wart.

I think that if we get rid of it, many of the ops could lose their context attribute.